fortnite_api/shop.py

```diff
@@ -1,266 +1,272 @@
+    layout_id: :class:`str`
+        The layout ID of this entry.
+    layout: :class:`ShopEntryLayout`
+        The layout of this entry.
+    dev_name: :class:`str`
+        The internal dev name of this entry.
+    offer_id: :class:`str`
+        The offer ID of this entry.
+    display_asset_path: Optional[:class:`str`]
+        The display asset path of this entry.
+    tile_size: :class:`str`
+        The tile size of this entry.
+    new_display_asset_path: :class:`str`
+        The new display asset path of this entry.
+    new_display_asset: :class:`ShopEntryNewDisplayAsset`
+        The new display asset of this entry.
+    br_items: List[:class:`CosmeticBr`]
+        The Battle Royale cosmetics in this entry.
+    tracks: List[:class:`CosmeticTrack`]
+        The tracks in this entry.
+    instruments: List[:class:`CosmeticInstrument`]
+        The instruments in this entry.
+    cars: List[:class:`CosmeticCar`]
+        The cars in this entry.
+    lego_kits: List[:class:`CosmeticLegoKit`]
+        The LEGO kits in this entry.
+    """
 
-    def __init__(self, data: dict):
-        self.offer_image_url: str = data['OfferImage']
-        self.fnm_texture_url: Optional[str] = data.get('FNMTexture')
-        self.image_background_url: Optional[str] = data.get('ImageBackground')
-        self.background_url: Optional[str] = data.get('Background')
-        self.raw_data: dict = data
+    __slots__: Tuple[str, ...] = (
+        'regular_price',
+        'final_price',
+        'bundle',
+        'banner',
+        'giftable',
+        'refundable',
+        'sort_priority',
+        'layout_id',
+        'layout',
+        'dev_name',
+        'offer_id',
+        'display_asset_path',
+        'tile_size',
+        'new_display_asset_path',
+        'new_display_asset',
+        'br_items',
+        'tracks',
+        'instruments',
+        'cars',
+        'lego_kits',
+    )
+
+    def __init__(self, *, data: Dict[str, Any], http: HTTPClientT) -> None:
+        self.regular_price: int = data['regularPrice']
+        self.final_price: int = data['finalPrice']
 
+        _bundle = data.get('bundle')
+        self.bundle: Optional[ShopEntryBundle[HTTPClientT]] = _bundle and ShopEntryBundle(data=_bundle, http=http)
 
-class BrShop:
-    """Represents a Battle Royale shop.
+        _banner = data.get('banner')
+        self.banner: Optional[ShopEntryBanner] = _banner and ShopEntryBanner(data=_banner)
+
+        self.giftable: bool = data['giftable']
+        self.refundable: bool = data['refundable']
+        self.sort_priority: int = data['sortPriority']
+        self.layout_id: str = data['layoutId']
+        self.layout: ShopEntryLayout[HTTPClientT] = ShopEntryLayout(data=data['layout'], http=http)
+        self.dev_name: str = data['devName']
+        self.offer_id: str = data['offerId']
+        self.display_asset_path: Optional[str] = data.get('displayAssetPath')
+        self.tile_size: str = data['tileSize']
```

Comment:
We can Create an enum for TileSize. Values are: Small, Normal, DoubleWide and TripleWide. There was also a case, where Epic got a type in the tile size and it was spelled "nomal" instead of normal. In-game it was shown properly as Normal width, so this might be the fallback.